x-pack/plugins/global_search_bar/public/components/search_bar.tsx

```diff
@@ -0,0 +1,189 @@
+const blurEvent = new FocusEvent('blur');
+
+export function SearchBar({ globalSearch, navigateToUrl }: Props) {
+  const isMounted = useMountedState();
+  const [searchValue, setSearchValue] = useState<string>('');
+  const [searchRef, setSearchRef] = useState<HTMLInputElement | null>(null);
+  const [options, _setOptions] = useState([] as EuiSelectableTemplateSitewideOption[]);
+  const isMac = navigator.platform.toLowerCase().indexOf('mac') >= 0;
+
+  const setOptions = useCallback(
+    (_options: GlobalSearchResult[]) => {
+      if (!isMounted()) return;
+
+      _setOptions([
+        ..._options.map((option) => ({
+          key: option.id,
+          label: option.title,
+          url: option.url,
+          ...(option.icon && { icon: { type: option.icon } }),
+          ...(option.type &&
+            option.type !== 'application' && { meta: [{ text: cleanMeta(option.type) }] }),
+        })),
+      ]);
+    },
+    [isMounted, _setOptions]
+  );
+
+  useDebounce(
+    () => {
+      let arr: GlobalSearchResult[] = [];
+      globalSearch(searchValue, {}).subscribe({
+        next: ({ results }) => {
+          if (searchValue.length > 0) {
+            arr = [...results, ...arr].sort((a, b) => {
+              if (a.score < b.score) return 1;
+              if (a.score > b.score) return -1;
+              return 0;
+            });
+            setOptions(arr);
+            return;
+          }
+
+          // if searchbar is empty, filter to only applications and sort alphabetically
+          results = results.filter(({ type }: GlobalSearchResult) => type === 'application');
+
+          arr = [...results, ...arr].sort((a, b) => {
+            const titleA = a.title.toUpperCase(); // ignore upper and lowercase
+            const titleB = b.title.toUpperCase(); // ignore upper and lowercase
+            if (titleA < titleB) return -1;
+            if (titleA > titleB) return 1;
+            return 0;
+          });
+
+          setOptions(arr);
+        },
+        error: () => {
+          // TODO #74430 - add telemetry to see if errors are happening
+          // Not doing anything on error right now because it'll either just show the previous
+          // results or empty results which is basically what we want anyways
+        },
+        complete: () => {},
+      });
+    },
+    250,
+    [searchValue]
+  );
+
+  const onKeyDown = (event: KeyboardEvent) => {
+    if (event.key === '/' && (isMac ? event.metaKey : event.ctrlKey)) {
+      if (searchRef) {
+        event.preventDefault();
+        searchRef.focus();
+      }
+    }
+  };
+
+  const onChange = (selected: EuiSelectableTemplateSitewideOption[]) => {
+    // @ts-ignore - ts error is "union type is too complex to express"
+    const { url } = selected.find(({ checked }) => checked === 'on');
+
+    if (/^https?:\/\//.test(url)) window.location.assign(url);
+    else {
+      navigateToUrl(url);
```

Comment:
`navigateToUrl` already handles external links, so I think you can remove this check and always delegate the navigation to the core API.